exercises/practice/palindrome-products/.articles/is-palindrome-check/content.md

```diff
@@ -0,0 +1,165 @@
+# Is number a palindrome? 
+
+There are multiple approaches to solve the palindrome product exercise. 
+Most of them require us to check if a given number is a palindrome. 
+
+## Approaches
+There are multiple ways to do it in C# and we will look at them in this article. 
+
+### String reversal
+
+Possibly the most obvious solution is to treat a number as a string of characters
+and reverse it. Let's have a look:
+
+```csharp
+using System.Linq;
+
+private static bool IsPalindrome(int number)
+{
+    var original = number.ToString();
+    var reversed = new String(original.Reverse().ToArray());
+    return original.Equals(reversed);
+}
+```
+
+If you come from some other languages you might be surprised by the above code.
+Unfortunately, C# doesn't come with a built in method to reverse a string. 
+We have to use `.Reverse()` extension method provided by 
+[Language-Integrated Query (LINQ)](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/linq/). However, it returns an iterator and so it has to be collected with `.ToArray()` 
+and converted into a `String` with `new String(char[])` constructor. 
+
+
+If this looks like a lot of code to you, it might be slightly shortened using 
+[string interpolation](https://learn.microsoft.com/en-us/dotnet/csharp/tutorials/string-interpolation)
+to convert a number into a text like in the example below. However, before you use it
+read the performance considerations below.
+
+```csharp
+using System.Linq;
+
+private static bool IsPalindrome(int number)
+=>  $"{number}".Equals(new String($"{number}".Reverse().ToArray()));
```

Comment:
This is more consistent with existing docs:

```suggestion
private static bool IsPalindrome(int number) =>
    $"{number}".Equals(new String($"{number}".Reverse().ToArray()));
```

Reply:
I had a quick look at recent style guides at [Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/coding-conventions), [dotnet project](https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md) and [Google](https://google.github.io/styleguide/csharp-style.html). Non of them provide clear guidance what to do in situations where `=>` operator is used to define a function where function has a single statement, but is too long for a single line. The common practice is to run with too long lines which make readability suffer. 

The style which is more common, as you suggest, in the documentation is to leave the `=>` hanging at the end of the line, but then it might be difficult to spot and quickly understand the meaning of the statement. The alternative is to start the next line with the operator. This is something I have recently picked up while mentoring the C# track. Some people out there use it to great effect. I also like how it falls in line with how we see the text. If you haven't seen it I do recommend [Seven Ineffective Coding Habits of Many Programmers by Kevlin Henney](https://www.youtube.com/watch?v=ZsHMHukIlJY), especially [the part about style](https://www.youtube.com/watch?v=ZsHMHukIlJY&t=1027s).

So, with some vagueness around the official style guides and emerging practice, perhaps we could improve here and over time help this become the majority option?